contrib/gitian-build.sh

```diff
@@ -1,406 +1,367 @@
+            ;;
+        # lxc
+        --lxc)
+            USE_LXC=1
             ;;
-        # Detach sign
-        --detach-sign)
-            signProg="true"
-            commitFiles=false
+        # docker
+        --docker)
+            USE_DOCKER=1
             ;;
-        # Commit files
-        --no-commit)
-            commitFiles=false
+        # URL
+        -u)
+            if [ -n "$2" ]; then
+                url=$2
+                shift
+            else
+                echo 'Error: "-u" requires an argument'
+                exit 1
+            fi
             ;;
         # Setup
         --setup)
             setup=true
             ;;
-	*)               # Default case: If no more options then break out of the loop.
-             break
+        --test)
+            test=true
+            ;;
+        *) # Default case: If no more options then break out of the loop.
+            break ;;
     esac
     shift
 done
 
-# Set up LXC
-if [[ $lxc = true ]]
-then
-    export USE_LXC=1
-    export LXC_BRIDGE=br0
-fi
+# Download specific file and verify hash
+function download_file () {
+    local filename=$(basename $1)
 
-# Check for OSX SDK
-if [[ ! -e "gitian-builder/inputs/MacOSX10.11.sdk.tar.gz" && $osx == true ]]
-then
-    echo "Cannot build for OSX, SDK does not exist. Will build for other OSes"
-    osx=false
-fi
+    if [ ! -f $filename ]; then
+        wget $1
+    fi
 
-# Get signer
-if [[ -n"$1" ]]
-then
-    SIGNER=$1
-    shift
-fi
+    # Verify file signature
+    echo "$2 $filename" | sha256sum -c --status
 
-# Get version
-if [[ -n "$1" ]]
-then
-    VERSION=$1
+    if [ $? != 0 ]; then
+        echo "$scriptName: Signature for $filename don't match."
+        exit 1
+    fi
+}
+
+function move_build_files() {
+    find build/out -type f -exec mv '{}' $outputDir/dogecoin-binaries/${VERSION}/ \;
+}
+
+function download_descriptor() {
+    if [[ ! $1 == 'test' ]]; then
+        uri="${url/github.com/raw.githubusercontent.com}"/"$2"/contrib/gitian-descriptors/gitian-"$1".yml
+        echo "Downloading descriptor ${1} ${uri}"
+        wget $uri -O gitian-"$1".yml || exit 1
+    else
+        # CI tests
+        cp ../ci/descriptor/"$1".yml gitian-"$1".yml || exit 1
+    fi
+}
+
+### Test configuration ###
+
+if [[ $test == true ]]; then
+    if [[ $commit == true ]]; then
+        VERSION="f80bfe9068ac1a0619d48dad0d268894d926941e"
+    else
+        VERSION="1.14.3"
+    fi
+    DESCRIPTORS=('test')
+    SIGN_DESCRIPTORS=()
     COMMIT=$VERSION
-    shift
+    SIGNER="signer"
+    SHA256SUM="0d519f6ade0e601617a7c44b764eeae35a8784070c3e44f542011956f1743459"
 fi
 
-# Check that a signer is specified
-if [[ $SIGNER == "" ]]
-then
-    echo "$scriptName: Missing signer."
-    echo "Try $scriptName --help for more information"
-    exit 1
+### Arguments checks ####
+
+echo "Using ${proc} CPU and ${mem} RAM"
+
+# Control the selection of a single virtualisation software
+if [ $(($USE_LXC + $USE_DOCKER)) -ge 2 ]; then
```

Comment:
Since KVM is not documented, could we please do an `elif` with `-lt 1` and throw an error too for now?

Reply:
Let's see your answer related to keep or remove KVM. In anticipation of the case where you want to remove it, do you want to use docker or lxc as default, or do we require from the user to specify it ?